src/main/java/mekanism/common/util/MinerUtils.java

```diff
@@ -28,67 +31,56 @@
         Block block = state.getBlock();
 
-        if (block.isAir(state, world, obj.getPos())) {
-            return Collections.EMPTY_LIST;
+        if (block.isAir(state, world, coord.getPos())) {
+            return Collections.emptyList();
         }
 
-        if (silk && (
-              block.canSilkHarvest(world, obj.getPos(), state, Mekanism.proxy.getDummyPlayer((WorldServer) world).get())
-                    || specialSilkIDs.contains(block))) {
-            List<ItemStack> ret = new ArrayList<>();
+        if (silk && (block.canSilkHarvest(world, coord.getPos(), state,
+              Mekanism.proxy.getDummyPlayer((WorldServer) world).get()) || specialSilkIDs.contains(block))) {
+            Object it = null;
             if (getSilkTouchDrop != null) {
                 try {
-                    Object it = getSilkTouchDrop.invoke(block, state);
-                    if (it instanceof ItemStack && !((ItemStack) it).isEmpty()) {
-                        ret.add((ItemStack) it);
-                    } else if (it instanceof ItemStack && ((ItemStack) it)
-                          .isEmpty()) {//silk touch drop is empty, fallback to grabbing an itemblock
-                        fallbackGetSilkTouch(block, state, ret);
-                    }
+                    it = getSilkTouchDrop.invoke(block, state);
                 } catch (InvocationTargetException | IllegalAccessException e) {
                     Mekanism.logger.error("Block.getSilkTouchDrop errored", e);
-                    fallbackGetSilkTouch(block, state, ret);
                 }
-            } else//fallback to old method
-            {
-                fallbackGetSilkTouch(block, state, ret);
             }
-
+            List<ItemStack> ret = new ArrayList<>();
+            if (it instanceof ItemStack && !((ItemStack) it).isEmpty()) {
+                ret.add((ItemStack) it);
+            } else {
+                //silk touch drop is empty or failed to call/find getSilkTouchDrop method
+                // Fallback to grabbing an itemblock
+                Item item = Item.getItemFromBlock(block);
+                if (item != Items.AIR) {
+                    ret.add(new ItemStack(item, 1, item.getHasSubtypes() ? block.getMetaFromState(state) : 0));
+                }
+            }
             if (ret.size() > 0) {
-                net.minecraftforge.event.ForgeEventFactory
-                      .fireBlockHarvesting(ret, world, obj.getPos(), state, 0, 1.0f, true, null);
+                ForgeEventFactory.fireBlockHarvesting(ret, world, coord.getPos(), state, 0, 1.0F, true, null);
                 return ret;
             }
         } else {
-            @SuppressWarnings("deprecation")
-            List<ItemStack> blockDrops = block.getDrops(world, obj.getPos(), state, 0);
+            NonNullList<ItemStack> blockDrops = NonNullList.create();
+            block.getDrops(blockDrops, world, coord.getPos(), state, 0);
```

Comment:
This needs to use the deprecated one, as if anyone overrides the old method it won't get caught.

Override = new one, Call = old one